dali/operators/generic/one_hot.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2017-2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_ONE_HOT_H_
+#define DALI_OPERATORS_GENERIC_ONE_HOT_H_
+
+#include <vector>
+
+#include "dali/pipeline/operator/operator.h"
+#include "dali/kernels/kernel_params.h"
+#include "dali/core/tensor_view.h"
+#include "dali/core/static_switch.h"
+
+#define ONE_HOT_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float, double)  // NOLINT
+
+namespace dali {
+
+namespace detail {
+template <typename Out, typename In, int ndims = 1>
+void DoOneHot(kernels::OutTensorCPU<Out, ndims> out,
+              kernels::InTensorCPU<In, ndims> in,
+              int depth, int on_value, int off_value) {
+  auto input = in.data;
+  auto output = out.data;
+  if (in.shape.sample_dim() == 1) {
+    for (int sample = 0; sample < in.shape[0]; ++sample) {
+      for (int i = 0; i < depth; ++i) {
+        if (i == static_cast<int>(input[sample])) {
+          output[sample * depth + i] = on_value;
+        } else {
+          output[sample * depth + i] = off_value;
```

Comment:
You're obviously indexing the output as 2D - I think it should be 2D; your operator output is (see SetupImpl).